src/js-module.ts

```diff
@@ -12,42 +12,31 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {ConvertedDocumentUrl} from './url-converter';
+import {ConvertedDocumentFilePath, ConvertedDocumentUrl, OriginalDocumentUrl} from './url-converter';
 
-export type ConversionOutput = DeleteFile|HtmlFile|JsModule;
-
-export interface DeleteFile {
-  readonly type: 'delete-file';
-  readonly url: ConvertedDocumentUrl;
-}
 
 export interface HtmlFile {
   readonly type: 'html-file';
-  readonly url: ConvertedDocumentUrl;
-
   readonly source: string;
 }
 
 export interface JsModule {
   readonly type: 'js-module';
-  /**
-   * Package-relative URL of the converted JS module.
-   */
-  readonly url: ConvertedDocumentUrl;
-
-  /**
-   * Converted source of the JS module.
-   */
   readonly source: string;
-
-  /**
-   * Set of exported names.
-   */
+  readonly exportedNamespaceMembers: ReadonlyArray<NamespaceMemberToExport>;
+  /** Set of exported names. */
   readonly es6Exports: ReadonlySet<string>;
+}
 
-  readonly exportedNamespaceMembers: ReadonlyArray<NamespaceMemberToExport>;
+export interface ConversionResult {
+  readonly originalUrl: OriginalDocumentUrl;
+  readonly convertedUrl: ConvertedDocumentUrl;
+  readonly convertedFilePath: ConvertedDocumentFilePath;
+  readonly keepOriginal?: boolean;
```

Comment:
Can you document this one? What are the implied side effects of this being true? What's the intended default?